src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  onElectTodoId: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onElectTodoId }) => {
+  const [user, setUser] = useState<User>();
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
```

Comment:
Create a handle function for this logic.
React Hook `useEffect `has a missing dependency